lib/adapters/xhr.js

```diff
@@ -142,14 +142,15 @@ module.exports = function xhrAdapter(config) {
     }
 
     // Handle progress if needed
-    if (typeof config.progress === 'function') {
-      if (config.method === 'post' || config.method === 'put') {
-        request.upload.addEventListener('progress', config.progress);
-      } else if (config.method === 'get') {
-        request.addEventListener('progress', config.progress);
-      }
+    if (typeof config.progressDownload === 'function') {
+      request.addEventListener('progress', config.progressDownload);
+    }
+
+    if (typeof config.progressUpload === 'function') {
+      request.upload.addEventListener('progress', config.progressUpload);
```

Comment:
It's not really clear which browsers support `request.upload` and when they did. It was included in the XHR2 spec so older browsers definitely won't support it. Will add a check.